web/app/js/components/util/TapUtils.jsx

```diff
@@ -135,34 +135,66 @@ const publicAddressToString = ipv4 => {
 /*
   display more human-readable information about source/destination
 */
-export const srcDstColumn = (display, labels, ResourceLink) => {
-  let podLink = (
-    <ResourceLink
-      resource={{ type: "pod", name: labels.pod, namespace: labels.namespace }}
-      linkText={"po/" + labels.pod} />
+const podLink = (labels, ResourceLink) => (
+  <ResourceLink
+    resource={{ type: "pod", name: labels.pod, namespace: labels.namespace }}
+    linkText={"po/" + labels.pod} />
+);
+
+const deployLink = (labels, ResourceLink) => (
+  <ResourceLink
+    resource={{ type: "deployment", name: labels.deployment, namespace: labels.namespace}}
+    linkText={"deploy/" + labels.deployment} />
+);
+
+const resourceSection = (ip, labels, ResourceLink) => {
+  return (
+    <React.Fragment>
+      <div>{ !labels.deployment ? null : deployLink(labels, ResourceLink) }</div>
+      <div>{ !labels.pod ? null : podLink(labels, ResourceLink) }</div>
```

Comment:
And similarly here, I think we should take into account resource type when generating the list of links in the popover. Or it might be cool to render a link for every label that exists, minus the namespace label. That way if we're viewing a pod, we see all of it's linkable owners in the popover.

Reply:
yeah! I like the idea of trying all of these!